package/Shaders/Water.hlsl

```diff
@@ -560,15 +564,15 @@ float3 GetWaterSpecularColor(PS_INPUT input, float3 normal, float3 viewDirection
 			reflectionColor = ReflectionTex.SampleLevel(ReflectionSampler, reflectionNormal.xy / reflectionNormal.ww, 0).xyz;
 		}
 
-#			if !defined(LOD) && NUM_SPECULAR_LIGHTS == 0 && !defined(VR)
+#			if NUM_SPECULAR_LIGHTS == 0 && !defined(VR)
 		if (PixelShaderDescriptor & _Cubemap) {
 			float2 ssrReflectionUv = GetDynamicResolutionAdjustedScreenPosition((DynamicResolutionParams2.xy * input.HPosition.xy) * SSRParams.zw + SSRParams2.x * normal.xy);
 			float4 ssrReflectionColor1 = SSRReflectionTex.Sample(SSRReflectionSampler, ssrReflectionUv);
 			float4 ssrReflectionColor2 = RawSSRReflectionTex.Sample(RawSSRReflectionSampler, ssrReflectionUv);
 			float4 ssrReflectionColor = lerp(ssrReflectionColor2, ssrReflectionColor1, SSRParams.y);
 
 			finalSsrReflectionColor = max(0, ssrReflectionColor.xyz);
-			ssrFraction = saturate(ssrReflectionColor.w * (SSRParams.x * distanceFactor));
+			ssrFraction = saturate(ssrReflectionColor.w * SSRParams.x * (1.0 - saturate(length(input.WPosition.xyz) / 320000)));
```

Comment:
Should we maybe move the hardcoded `320000` to constants.hlsli? To make it clearer what it does as well

Reply:
It's an arbitrary value. Just that around further than that you see issues on the sea of ghosts